benches/oneshot/ipa.rs

```diff
@@ -9,7 +9,7 @@ async fn main() -> Result<(), Error> {
     let mut config = TestWorldConfig::default();
     config.gateway_config.send_buffer_config.items_in_batch = 1;
     config.gateway_config.send_buffer_config.batch_count = 1000;
-    let world = TestWorld::new_with(config);
+    let world = TestWorld::new_with(config).await;
```

Comment:
lots of files changed just because `TestWorld` constructor is `async` now. It must be `async` because `Transport` is capable of serving multiple queries running in parallel (i.e. there must be an event loop somewhere) and `Gateway` needs to subscribe and wait until `Transport` acknowledges the request to route query-specific commands to that gateway